addons/website/models/website_visitor.py

```diff
@@ -235,6 +240,11 @@ def _create_visitor(self):
             'country_id': country_id,
             'website_id': request.website.id,
         }
+
+        tz = self._get_visitor_timezone()
+        if tz:
+            vals['timezone'] = tz
+
         if not self.env.user._is_public():
             vals['partner_id'] = self.env.user.partner_id.id
             vals['name'] = self.env.user.partner_id.name
```

Comment:
If we have no tz from cookies, we could sync it with the user one is user is not public :
```
if not vals['tz']: 
    vals['tz'] = self.env.user.tz
```